raiden/tests/integration/transfer/test_refundtransfer.py

```diff
@@ -177,6 +193,31 @@ def test_refund_transfer(raiden_chain, number_of_nodes, token_addresses, deposit
         app2, deposit + amount_path + amount_drain, [],
     )
 
+    # Additional checks for LockExpired causing nonce mismatch after refund transfer:
+    # https://github.com/raiden-network/raiden/issues/3146#issuecomment-447378046
+    # At this point make sure that the initiator has not deleted the payment task
+    assert state_from_raiden(app0.raiden).payment_mapping.secrethashes_to_task[secrethash]
```

Comment:
Nitpick: i would opt for checking for `X in Y` rather than this, because this would fail with a KeyError while the `in` check would return False.